### ISSUE TITLE
Migrate to `eslint.config.js`

### ISSUE THREAD

**Issue by jasongitmail** [sveltejs/kit#10742]
### Describe the problem

SvelteKit currently uses `.eslintrc.cjs`, the outgoing format.

### Describe the proposed solution

Switch to `eslint.config.js` and remove `.eslintignore` and `.eslintrc.cjs`.

Starting in v9, ESLint will use one config file named `eslint.config.js`, and it's possible to remove `.eslintignore` and [specify ignored file patterns within the config](https://eslint.org/docs/latest/use/configure/ignore#ignorepatterns-in-config-files).  

Good info: https://eslint.org/blog/2022/08/new-config-system-part-2/

Comes with a bunch of benefits--including reducing config pollution within a project--and its easier to extend with custom user configs.

### Alternatives considered

Will be required in v9 ESLint.

### Importance

nice to have

### Additional Information

Adding the issue to track it, so I can update existing projects similarly.

**Comment by eltigerchino** [comment]
I think we'll need to wait for the typescript-eslint parser to support flat configs before we can go ahead with this https://github.com/typescript-eslint/typescript-eslint/issues/7694#issuecomment-1854655034

**Comment by victormihalache** [comment]
Will such change be something the migration script will do anything about? At least emit a notice of sorts?

**Comment by lts20050703** [comment]
Any updates on this? Eslint v9 just got released a few hours ago

**Comment by WarningImHack3r** [comment]
I just tried migrating by myself [EmeraldHQ/Website@`29f3a15` (#227)](https://github.com/EmeraldHQ/Website/pull/227/commits/29f3a15d29f772022bfa8ff2abd46d9c3ceb4511#diff-a32a0887ed9d1d707bbb3b845b7df7fd40e673c47e7b60a3ebd896b68d3b8839)
But it seems that [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte) isn't ready yet or that something is wrong in my attempt: I got lint errors (related to TypeScript I guess?), not sure why
